pkg/orgIdColumnPopulator/main.go

```diff
@@ -0,0 +1,178 @@
+
+	var uniqueAccounts int64
+	var rowsUpdated int64
+
+	statement, err := buildAccountsQuery(database, table, accountColumn, orgIdColumn, nullOrgIdPlaceholder, batchSize)
+	if err != nil {
+		return uniqueAccounts, rowsUpdated, err
+	}
+	defer statement.Close()
+
+	for {
+
+		eans, err := readAccountsFromDatabase(ctx, database, statement, batchSize, dbOperationTimeout)
+		if err != nil {
+			return uniqueAccounts, rowsUpdated, err
+		}
+
+		log.Printf("Read %d unique accounts from table %s\n", len(eans), table)
+
+		if len(eans) == 0 {
+			return uniqueAccounts, rowsUpdated, nil
+		}
+
+		translationResults, err := batchTranslator.EANsToOrgIDs(ctx, eans)
+		if err != nil {
+			return uniqueAccounts, rowsUpdated, err
+		}
+
+		for _, results := range translationResults {
+
+			uniqueAccounts++
+
+			if results.Err != nil {
+				log.Printf("Skipping translation of EAN (%s) to OrgID (%s) due to error: %s\n", results.EAN, results.OrgID, results.Err)
+				continue
+			}
+
+			if results.EAN == nil {
+				log.Printf("Skipping nil EAN\n")
+				continue
+			}
+
+			updatedRowsCount, err := updateOrgIdInDatabase(ctx, database, table, accountColumn, orgIdColumn, nullOrgIdPlaceholder, *results.EAN, results.OrgID, dbOperationTimeout)
+			if err != nil {
+				return uniqueAccounts, rowsUpdated, err
+			}
+
+			if updatedRowsCount == 0 {
+				// I can't really think of why this would happen...but I really don't want
+				// this thing to just run forever...
+				log.Println("FAILED TO UPDATED ANY ROWS...Looks like we are done!!")
+				return uniqueAccounts, rowsUpdated, fmt.Errorf("Failed to update any rows.")
+			}
+
+			rowsUpdated = rowsUpdated + updatedRowsCount
+		}
+	}
+
+	return uniqueAccounts, rowsUpdated, nil
+}
+
+func buildAccountsQuery(database *sql.DB, table, accountColumn, orgIdColumn blessedIdentifier, nullOrgIdPlaceholder blessedLiteral, batchSize int) (*sql.Stmt, error) {
+
+	selectQuery := fmt.Sprintf("SELECT %s FROM %s WHERE %s IS NULL OR %s = %s GROUP BY %s LIMIT %d",
+		accountColumn,
+		table,
+		orgIdColumn,
+		orgIdColumn,
+		nullOrgIdPlaceholder,
+		accountColumn,
+		batchSize)
+
+	statement, err := database.Prepare(selectQuery)
+	if err != nil {
+		return nil, fmt.Errorf("error reading accounts from the database - prepare failed %w", err)
+	}
+
+	return statement, nil
+}
+
+func readAccountsFromDatabase(ctx context.Context, database *sql.DB, statement *sql.Stmt, batchSize int, timeout int) ([]string, error) {
+
+	ctx, cancel := context.WithTimeout(ctx, time.Duration(timeout)*time.Second)
+	defer cancel()
+
+	accountList := make([]string, 0, batchSize)
+
+	rows, err := statement.QueryContext(ctx)
+	if err != nil {
+		return accountList, fmt.Errorf("error reading accounts from the database - query failed %w", err)
+	}
+	defer rows.Close()
+
+	for rows.Next() {
+		var account string
+
+		if err := rows.Scan(&account); err != nil {
+			return accountList, fmt.Errorf("error reading accounts from the database - scan failed %w", err)
```

Comment:
Hey @dehort! In RBAC, we'll run into an issue where we know we'll have one row (our `public` tenant) which doesn't have an `account_number`. It will fail/return on this and not process subsequent rows.

Could we have this not kill the iteration, and just log the collective errors, or, it looks like we could also do something like this to make sure that `NULL` values would be effectively skipped:

```go
for rows.Next() {
        var account sql.NullString
        ...
        ...
        accountList = append(accountList, account.String)
}
```

Reference: https://pkg.go.dev/database/sql#NullString

By the way, this is awesome!

Reply:
Oh, that would work as well! :) Happy to test again when you make an update, but I did test the `sql.NullString` locally and that worked.

Reply:
@coderbydesign mind to give it another test?